src/test/java/edu/cornell/kfs/concur/batch/service/impl/ConcurRequestV4ServiceImplTest.java

```diff
@@ -0,0 +1,754 @@
+                RequestV4DetailFixture.PENDING_EXTERNAL_VALIDATION_TEST_REQUEST_JANE_DOE,
+                RequestV4DetailFixture.PENDING_EXTERNAL_VALIDATION_INVALID_TEST_REQUEST_JANE_DOE,
+                RequestV4DetailFixture.SENTBACK_INVALID_TEST_REQUEST_JANE_DOE,
+                RequestV4DetailFixture.PENDING_COST_APPROVAL_REGULAR_REQUEST_BOB_SMITH,
+                RequestV4DetailFixture.PENDING_EXTERNAL_VALIDATION_INVALID_REGULAR_REQUEST_BOB_SMITH,
+                RequestV4DetailFixture.PENDING_EXTERNAL_VALIDATION_REGULAR_REQUEST_BOB_SMITH,
+                RequestV4DetailFixture.NOT_SUBMITTED_REGULAR_REQUEST_BOB_SMITH
+        ).map(Arguments::of);
+    }
+
+    @Test
+    void testParsingOfTestUsersParameter() throws Exception {
+        Map<String, String> expectedMappings = buildExpectedUserIdMappings(
+                RequestV4PersonFixture.JOHN_TEST, RequestV4PersonFixture.JANE_DOE,
+                RequestV4PersonFixture.TEST_MANAGER, RequestV4PersonFixture.TEST_APPROVER);
+        assertParameterDerivedTestUserMapContainsExpectedEntries(expectedMappings);
+        
+        String newTestUsersParameterValue = buildTestUsersParameterValue(
+                RequestV4PersonFixture.BOB_SMITH, RequestV4PersonFixture.TEST_MANAGER);
+        overrideParameter(ConcurParameterConstants.REQUEST_V4_TEST_USERS, newTestUsersParameterValue);
+        
+        expectedMappings = buildExpectedUserIdMappings(
+                RequestV4PersonFixture.BOB_SMITH, RequestV4PersonFixture.TEST_MANAGER);
+        assertParameterDerivedTestUserMapContainsExpectedEntries(expectedMappings);
+    }
+
+    private Map<String, String> buildExpectedUserIdMappings(RequestV4PersonFixture... users) {
+        return Stream.of(users).collect(Collectors.toUnmodifiableMap(
+                userFixture -> userFixture.id, RequestV4PersonFixture::getNameForParameterEntry));
+    }
+
+    private void assertParameterDerivedTestUserMapContainsExpectedEntries(Map<String, String> expectedMappings) {
+        Map<String, String> actualMappings = requestV4Service.getRequestV4TestUserMappingsFromParameter();
+        assertNotNull(actualMappings, "The test users map should not have been null");
+        assertEquals(expectedMappings, actualMappings, "Wrong test user entries in map");
+    }
+
+    @ParameterizedTest
+    @ValueSource(strings = {
+            KFSConstants.EMPTY_STRING,
+            KFSConstants.BLANK_SPACE,
+            "Name1=",
+            "=Value1",
+            "Name1=Value1;Name2=;Name3=Value3",
+            "Name1=Value1;Name2=Value2;=Value3"
+    })
+    void testParsingFailureForInvalidTestUserParameterValue(String invalidValue) {
+        overrideParameter(ConcurParameterConstants.REQUEST_V4_TEST_USERS, invalidValue);
+        assertThrows(RuntimeException.class, () -> requestV4Service.getRequestV4TestUserMappingsFromParameter(),
+                "The parsing of the test users parameter should have failed when its value is blank or malformed");
```

Comment:
Should the value of invalidValue be included in this message, it might help if this test breaks in the future.

Reply:
Good point, I'll add the value to the failure message.